get_ci_vm/entrypoint.sh

```diff
@@ -0,0 +1,417 @@
+DESTDIR=""
+# full URL to project's "upstream" repo.
+UPSTREAM_REPO=""
+#
+# Optional:
+#
+# path where setup script persists it's env. vars
+CI_ENVFILE=""
+#
+# 'gcevm' instances type required:
+#
+# local gcloud sdk configuration name destinct from other repos.
+GCLOUD_CFG=""
+# GCE zone to create instances in (may be passed in to container)
+GCLOUD_ZONE="${GCLOUD_ZONE}"
+# number of CPUs to create instance with
+GCLOUD_CPUS=""
+# memory quantity to assign to instance (accepts unit suffix)
+GCLOUD_MEMORY=""
+# size of disk to attach (minimum is 200)
+GCLOUD_DISK=""
+# GCE Project ID to create instances in
+GCLOUD_PROJECT=""
+# GCE Project ID where images are stored
+GCLOUD_IMGPROJECT=""
+
+status() {
+    local message="$1"
+    local red="\e[1;31m"
+    local yel="\e[1;32m"
+    local nor="\e[0m"
+    (
+        echo ""  # Message should stand out vs any prior output
+        echo -e "${red}#${nor} ${yel}${message}${nor}"
+    ) > /dev/stderr
+}
+
+# Each repository behaves slightly differently, confirm
+# compatibility before loading repo-specific settings
+supports_apiv1() {
+    req_env_vars REPO_HACK_SCRIPT
+    if [[ ! -r "$REPO_HACK_SCRIPT" ]]; then
+        die "Can't find '$REPO_HACK_SCRIPT' in source repository."
+    elif grep -Eq '^# get_ci_vm APIv1' "$REPO_HACK_SCRIPT"; then
+        return 0
+    fi
+    return 1
+}
+
+init() {
+    status "Initializing get_ci_vm"
+    # These are passed in by hack/get_ci_vm.sh script
+    req_env_vars NAME SRCDIR
+
+    if [[ "$NAME" == "root" ]]; then
+        die "Running as root not supported, use your regular user account for identification/auditing purposes"
+    fi
+
+    _TMPDIR=$(mktemp -d -p '' get_ci_vm_XXXXXX.tmp)
+    # Several setup functions/commands expect this to be an absolute path
+    if [[ "${DESTDIR:0:1}" != "/" ]]; then
+        DESTDIR="/$DESTDIR"
+    fi
+
+    REPO_HACK_SCRIPT="$SRCDIR/hack/get_ci_vm.sh"
+    if supports_apiv1; then
+        # Dump+Source needed to support in-line comments
+        bash "$REPO_HACK_SCRIPT" --config > $_TMPDIR/apiv1.sh
+        # shellcheck disable=SC1090
+        source $_TMPDIR/apiv1.sh
+        req_env_vars DESTDIR UPSTREAM_REPO
+        # CI_ENVFILE is optional.
+    else
+        die "Repository hack/get_ci_vm.sh not compatible with this container image."
+    fi
+}
+
+get_inst_image() {
+    local cirrus_tasks
+    local _output
+    local -a type_image
+    status "Obtaining task listing from repository .cirrus.yml"
+    req_env_vars SRCDIR
+    cirrus_tasks=$(cirrus-ci_env.py --list "$SRCDIR/.cirrus.yml")
+
+    if [[ -z "$CIRRUS_TASK" ]]; then
+        die "Usage: hack/get_ci_vm.sh <task name | 'list'>
+       Note: Quoting the parameter is not required
+"
+    elif [[ "$CIRRUS_TASK" == "list" ]]; then
+        msg "$cirrus_tasks"
+        exit 0
+    elif ! grep -q "$CIRRUS_TASK"<<<"$cirrus_tasks"; then
+        die "Unknown .cirrus.yml task name '$CIRRUS_TASK', use 'list' to show valid names."
```

Comment:
`die "Unknown .cirrus.yml task name '$CIRRUS_TASK', use '--list' to show valid names."`